pages/settings/admin/tasks.vue

```diff
@@ -0,0 +1,248 @@
+                  "
+                  v-text="$t('settings.tasks.starting')"
+                />
+                <v-card-text
+                  v-else-if="
+                    taskStatus[task.Id] === running &&
+                    !$vuetify.breakpoint.smAndDown
+                  "
+                  v-text="$t('settings.tasks.running')"
+                />
+                <v-card-text
+                  v-else-if="
+                    taskStatus[task.Id] === cancelling &&
+                    !$vuetify.breakpoint.smAndDown
+                  "
+                  v-text="$t('settings.tasks.cancelling')"
+                />
+              </span>
+              <v-card-actions class="ml-auto mr-2">
+                <v-btn icon disabled>
+                  <v-icon>mdi-pencil</v-icon>
+                </v-btn>
+                <v-btn
+                  icon
+                  :color="taskStatus[task.Id] === running ? 'red' : 'primary'"
+                  :disabled="taskStatus[task.Id] === cancelling"
+                  @click="onClick(task.Id)"
+                >
+                  <v-icon v-if="taskStatus[task.Id] === idle">mdi-play</v-icon>
+                  <v-icon v-else>mdi-stop</v-icon>
+                </v-btn>
+              </v-card-actions>
+            </div>
+          </v-card>
+        </div>
+      </v-list>
+    </template>
+  </settings-page>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { TaskInfo, TaskState } from '@jellyfin/client-axios';
+import { v4 as uuidv4 } from 'uuid';
+import { groupBy } from 'lodash';
+import { mapActions } from 'vuex';
+
+interface GroupedTasks {
+  [k: string]: TaskInfo[];
+}
+
+export default Vue.extend({
+  middleware: 'adminMiddleware',
+  async asyncData({ $api }) {
+    const rawTasks = (await $api.scheduledTasks.getTasks()).data;
+
+    rawTasks.forEach((task) => {
+      if (task.IsHidden) {
+        rawTasks.splice(rawTasks.indexOf(task));
+      }
+    });
+
+    rawTasks.sort((a: TaskInfo, b: TaskInfo) => {
+      if (a.Category && b.Category) {
+        if (a.Category < b.Category) {
+          return -1;
+        }
+        if (a.Category > b.Category) {
+          return 1;
+        }
+      }
+      return 0;
+    });
+
+    const allTasks = groupBy(rawTasks, 'Category');
+    const taskStatus = {} as { [k: string]: TaskState };
+    const taskProgresses = {} as { [k: string]: number };
+    rawTasks.forEach((task) => {
+      if (task.Id) {
+        taskStatus[task.Id] = task.State as TaskState;
+        if (task.CurrentProgressPercentage) {
+          taskProgresses[task.Id] = task.CurrentProgressPercentage;
+        } else {
+          taskProgresses[task.Id] = 0;
+        }
+      }
+    });
+
+    return { allTasks, taskStatus, taskProgresses };
+  },
+  data() {
+    const uuid = uuidv4();
+    return {
+      uuid,
+      idle: TaskState.Idle,
+      cancelling: TaskState.Cancelling,
+      running: TaskState.Running,
```

Comment:
I think it would be better to use TaskState.Idle where it is needed, instead of this.idle